File: 11689.cpp
/*a536: 11689 - Soda Surpler*/
#include <iostream>
using namespace std;

int main(){
    int n,a1,a2,b;
    cin>>n;

    for(int i=0;i<n;i++){
        cin>>a1>>a2>>b;
        int a=a1+a2;
        int c=0;
        int r=0;
        while(a/b!=0){
            c += a/b;
            r = a%b;
            a /= b;
            a += r;
        }
        cout<<c<<endl;
        
    }
    return 0;
}

File: 11689.py
'''a536: 11689 - Soda Surpler'''

n = input()

for i in range(int(n)):
    s = input()
    d = s.split(' ')
    a = int(d[0]) + int(d[1])
    b = int(d[2])
    c = 0
    r = 0
    while a//b != 0:
        c += a//b
        
        r = a%b
        a = a//b
        
        a = a + r
    print(c)
